src/scripts/main.js

```diff
@@ -1,11 +1,22 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener(eventName, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    });
+  });
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const div = document.createElement('div');
+
+  div.classList.add('message');
+
+  div.textContent = message;
+  document.body.appendChild(div);
```

Comment:
The `printMessage` function creates a new `div` element every time it is called. If this function is called frequently, it could lead to a large number of elements being added to the DOM. Consider reusing a single element or limiting the number of elements created to improve performance.